### PR TITLE
API: Proto for deploys

### PR DESCRIPTION
**Pull request by smonero**
### Description

It would be nice to have a protobuf for having a deployment service.

### GitHub Issue

Fixes #

### CODE REVIEW

File: api/deploys/v1/deploys.proto [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+message DeploymentOptions {
+  bool emergency_rollout = 1;
+  bool skip_deploy_checks = 2;
+  string initiating_user = 3;
+}
+
+message Deploy {
+  repeated Environment environments = 1 [ (validate.rules).repeated = {min_items : 1} ];
+  string project_name = 2 [ (validate.rules).string = {min_bytes : 1} ];
```
Comment (scarlettperry) [inline api/deploys/v1/deploys.proto:38]:
```suggestion
  string project = 2 [ (validate.rules).string = {min_bytes : 1} ];
```

File: api/deploys/v1/deploys.proto [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  }
+}
+
+message TriggerManualDeployRequest {
+  Deploy deploy_intent = 1;
+}
+
+message TriggerManualDeployResponse {
+  int64 deploy_job_id = 1;
+}
```
Comment (scarlettperry) [inline api/deploys/v1/deploys.proto:57]: nit, new line

File: api/deploys/v1/deploys.proto [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  DEVELOPMENT = 1;
+  STAGING = 2;
+  PRODUCTION = 3;
+}
+
+message DeploymentOptions {
+  bool emergency_rollout = 1;
+  bool skip_deploy_checks = 2;
+  string initiating_user = 3;
+}
```
Comment (scarlettperry) [inline api/deploys/v1/deploys.proto:34]: for OSS, I wonder if this should follow the pattern of the Project [proto](https://github.com/lyft/clutch/blob/main/api/core/project/v1/project.proto#L17-L18) where we use `map<string, google.protobuf.Value>`.

File: api/deploys/v1/deploys.proto [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+message Deploy {
+  repeated Environment environments = 1 [ (validate.rules).repeated = {min_items : 1} ];
+  string project_name = 2 [ (validate.rules).string = {min_bytes : 1} ];
+  string message = 3 [ (validate.rules).string = {min_bytes : 1} ];
+  string revision_sha = 4 [ (validate.rules).string.pattern = "[0-9a-f]{40}" ];
+
+  // Target could also be called Job or job name.
+  string target = 5 [ (validate.rules).string = {min_bytes : 1} ];
```
Comment (scarlettperry) [inline api/deploys/v1/deploys.proto:43]: if we're allowing users to specify multiple environments, then should this allow for more than 1 target?

File: api/deploys/v1/deploys.proto [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    DeploymentOptions deploy_options = 7;
+  }
+}
+
+message TriggerManualDeployRequest {
+  Deploy deploy_intent = 1;
+}
+
+message TriggerManualDeployResponse {
+  int64 deploy_job_id = 1;
```
Comment (scarlettperry) [inline api/deploys/v1/deploys.proto:56]:
```suggestion
  int64 job_id = 1;
```

**Review by scarlettperry — COMMENTED** [review]
some initial comments

File: api/deploys/v1/deploys.proto (hunk 1 above)
Comment (smonero) [reply api/deploys/v1/deploys.proto:38]: originally it was `project` but derek wanted it to be `project_name` https://github.com/lyft/clutch-private/pull/1699#discussion_r958775771

**Pull request closed without merging**